File: 3332-minimum-operations-to-exceed-threshold-value-ii/3332-minimum-operations-to-exceed-threshold-value-ii.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        minHeap=nums.copy()
        heapify(minHeap)
        count=0
        while minHeap[0]<k:
            x=heapq.heappop(minHeap)
            y=heapq.heappop(minHeap)
            a=2*x+y
            heapq.heappush(minHeap,a)
            count+=1
        return count

File: 3332-minimum-operations-to-exceed-threshold-value-ii/3332-minimum-operations-to-exceed-threshold-value-ii.java
class Solution {
    public int minOperations(int[] nums, int k) {
        PriorityQueue <Long> pq=new PriorityQueue<>();
        for(int i=0;i<nums.length;i++){
            pq.add((long)nums[i]);
        }
        int c=0;
        while(pq.peek()<k){
            long min1=pq.poll();
            if(!pq.isEmpty()){
            long min2=pq.poll();
            long a=(min1*2)+min2;
            pq.add(a);
            c++;
            }
            else{
                break;
            }
        }
        return c;
    }
}